lib/pool.js

```diff
@@ -6,24 +6,16 @@
 
 const P = require('./promise')
 const Poolee = require('poolee')
+const url = require('url')
 
-function parseUrl(url) {
-  var match = /([a-zA-Z]+):\/\/(\S+)/.exec(url)
-  if (match) {
-    return {
-      protocol: match[1],
-      host: match[2]
-    }
-  }
-  throw new Error('url is invalid: ' + url)
-}
-
-function Pool(url, options = {}) {
-  var parsedUrl = parseUrl(url)
-  var protocol = require(parsedUrl.protocol)
+function Pool(uri, options = {}) {
+  const parsed = url.parse(uri)
+  const {protocol, host} = parsed
+  const protocolModule = require(protocol.slice(0, -1))
```

Comment:
I'd feel a lot more comfortable with this if we imported them all the modules up-front into a map like:

```
PROTOCOL_MODULES = {
  'http': require('http'),
  'https': require('https'),
}
```

And then looked up the module in the map at run-time as `PROTOCOL_MODULES[protocol.slice(0, -1))]`.

Blindly `require()`ing based on a value loaded dynamically at runtime scares me; the current input strings are trusted values taken from config, but future us could naively start using `Pool` with values taken from user input without realizing it has this behaviour.

Reply:
Yeah good suggestion, I should have added this :P